javaslang/src/main/java/javaslang/collection/CharSeq.java

```diff
@@ -2132,16 +2132,13 @@ public CharSeq replace(CharSequence target, CharSequence replacement) {
      */
     public Seq<CharSeq> split(String regex, int limit) {
         final String[] javaStrings = back.split(regex, limit);
-        final IndexedSeq<CharSeq> result = Vector.empty();
-
-
-        for (int i = 0; i < javaStrings.length; i++) {
-            result.append(CharSeq.of(javaStrings[i]));
-
+        final CharSeq[] strings = new CharSeq[javaStrings.length];
```

Comment:
You might want to do it in a functional way instead.
The error in your previous code is probably that you the result of the append is ignored.
Please only commit, when all tests are passing.

```Java
public Seq<CharSeq> split(String regex, int limit) {
    Array<String> wrap = Array.wrap(back.split(regex, limit));
    return wrap.map(CharSeq::of);
}
```